app/src/main/java/com/github/se/signify/model/authentication/MockAuthService.kt

```diff
@@ -0,0 +1,88 @@
+@file:Suppress("DEPRECATION")
+
+package com.github.se.signify.model.authentication
+
+import android.content.Context
+import android.content.Intent
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.common.api.ApiException
+import com.google.firebase.auth.AuthResult
+
+/**
+ * `MockAuthService` is a mock implementation of the `AuthService` interface, designed for testing
+ * purposes. It provides fake authentication logic to simulate user sign-in, sign-out, and other
+ * authentication-related functionalities without requiring external services like Firebase.
+ */
+class MockAuthService : AuthService {
+
+  /**
+   * Simulates signing in a user with a Google ID token.
+   *
+   * @param idToken The Google ID token for the user.
+   * @return Always returns `true` to indicate a successful sign-in.
+   */
+  override suspend fun signInWithGoogle(idToken: String): Boolean {
+    return true
+  }
+
+  /**
+   * Simulates signing out the current user.
+   *
+   * @return Always returns `true` to indicate a successful sign-out.
+   */
+  override suspend fun signOut(): Boolean {
+    return true
+  }
+
+  /**
+   * Simulates retrieving an Intent for initiating the Google sign-in process.
+   *
+   * @param context The context from which the Intent is requested.
+   * @param token The web client ID token used for authentication.
+   * @return A mocked Intent for the Google sign-in process.
+   */
+  override fun getSignInIntent(context: Context, token: String): Intent {
```

Comment:
In the MockAuthService, this function simulates this process without interacting with real Google services. It's primarily used for testing purposes where you don’t want to depend on actual Google Sign-In operations.
Because doing so will need to launch an Intent where you have to enter the password/e-mail